src/modules/bulletproofs/bulletproofs_util.h

```diff
@@ -0,0 +1,52 @@
+/**********************************************************************
+ * Copyright (c) 2020 Andrew Poelstra                                 *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef _SECP256K1_MODULE_BULLETPROOFS_UTIL_
+#define _SECP256K1_MODULE_BULLETPROOFS_UTIL_
+
+#include "field.h"
+#include "group.h"
+#include "hash.h"
+
+/* Outputs a pair of points, amortizing the parity byte between them
+ * Assumes both points' coordinates have been normalized.
+ */
+static void secp256k1_bulletproofs_serialize_points(unsigned char *output, const secp256k1_ge *lpt, const secp256k1_ge *rpt) {
+    output[0] = (secp256k1_fe_is_odd(&lpt->y) << 1) + secp256k1_fe_is_odd(&rpt->y);
+    secp256k1_fe_get_b32(&output[1], &lpt->x);
+    secp256k1_fe_get_b32(&output[33], &rpt->x);
+}
+
+/* Initializes SHA256 with fixed midstate. This midstate was computed by applying
+ * SHA256 to SHA256("Bulletproofs_pp/v0/commitment")||SHA256("Bulletproofs_pp/v0/commitment").
+ */
+static void secp256k1_bulletproofs_sha256_tagged_commitment(secp256k1_sha256 *sha) {
+    secp256k1_sha256_initialize(sha);
+    sha->s[0] = 0x52fc8185ul;
+    sha->s[1] = 0x0e7debf0ul;
+    sha->s[2] = 0xb0967270ul;
+    sha->s[3] = 0x6f5abfe1ul;
+    sha->s[4] = 0x822bdec0ul;
+    sha->s[5] = 0x36db8beful;
+    sha->s[6] = 0x03d9e1f1ul;
+    sha->s[7] = 0x8a5cef6ful;
+
+    sha->bytes = 64;
```

Comment:
It will be good to add a test that compares the output of `sha256_finalize` on a SHA object initialized using this function to the output of `sha256_finalize` on a plain SHA object with the proper prefix written. This will ensure that the constants here are correct.